Jack/Gen.fs

```diff
@@ -434,6 +434,10 @@ module Gen =
     let float : Gen<float> =
         double |> map float
 
+    let guid : Gen<System.Guid> =
+        gen { let! bs = byte |> array' 16 16
+              return System.Guid bs }
```

Comment:
I hope so, according to [the constructor overload](https://msdn.microsoft.com/en-us/library/90ck37x3(v=vs.110).aspx?cs-save-lang=1&cs-lang=fsharp#code-snippet-1) that I've picked. It takes a 16-element byte array containing values with which to initialize the GUID.

```
> 
=== Outcome ===
0d05090c-040f-191d-0a13-12010d0c1601
=== Shrinks ===
0d050900-040f-191d-0a13-12010d0c1601
0d050906-040f-191d-0a13-12010d0c1601
0d050909-040f-191d-0a13-12010d0c1601
0d05090b-040f-191d-0a13-12010d0c1601
0d05000c-040f-191d-0a13-12010d0c1601
0d05050c-040f-191d-0a13-12010d0c1601
0d05070c-040f-191d-0a13-12010d0c1601
0d05080c-040f-191d-0a13-12010d0c1601
0d00090c-040f-191d-0a13-12010d0c1601
0d03090c-040f-191d-0a13-12010d0c1601
0d04090c-040f-191d-0a13-12010d0c1601
0005090c-040f-191d-0a13-12010d0c1601
0705090c-040f-191d-0a13-12010d0c1601
0a05090c-040f-191d-0a13-12010d0c1601
0c05090c-040f-191d-0a13-12010d0c1601
0d05090c-0400-191d-0a13-12010d0c1601
0d05090c-0408-191d-0a13-12010d0c1601
0d05090c-040c-191d-0a13-12010d0c1601
0d05090c-040e-191d-0a13-12010d0c1601
0d05090c-000f-191d-0a13-12010d0c1601
0d05090c-020f-191d-0a13-12010d0c1601
0d05090c-030f-191d-0a13-12010d0c1601
0d05090c-040f-1900-0a13-12010d0c1601
0d05090c-040f-190f-0a13-12010d0c1601
0d05090c-040f-1916-0a13-12010d0c1601
0d05090c-040f-191a-0a13-12010d0c1601
0d05090c-040f-191c-0a13-12010d0c1601
0d05090c-040f-001d-0a13-12010d0c1601
0d05090c-040f-0d1d-0a13-12010d0c1601
0d05090c-040f-131d-0a13-12010d0c1601
0d05090c-040f-161d-0a13-12010d0c1601
0d05090c-040f-181d-0a13-12010d0c1601
0d05090c-040f-191d-0013-12010d0c1601
0d05090c-040f-191d-0513-12010d0c1601
0d05090c-040f-191d-0813-12010d0c1601
0d05090c-040f-191d-0913-12010d0c1601
0d05090c-040f-191d-0a00-12010d0c1601
0d05090c-040f-191d-0a0a-12010d0c1601
0d05090c-040f-191d-0a0f-12010d0c1601
0d05090c-040f-191d-0a11-12010d0c1601
0d05090c-040f-191d-0a12-12010d0c1601
0d05090c-040f-191d-0a13-00010d0c1601
0d05090c-040f-191d-0a13-09010d0c1601
0d05090c-040f-191d-0a13-0e010d0c1601
0d05090c-040f-191d-0a13-10010d0c1601
0d05090c-040f-191d-0a13-11010d0c1601
0d05090c-040f-191d-0a13-12000d0c1601
0d05090c-040f-191d-0a13-1201000c1601
0d05090c-040f-191d-0a13-1201070c1601
0d05090c-040f-191d-0a13-12010a0c1601
0d05090c-040f-191d-0a13-12010c0c1601
0d05090c-040f-191d-0a13-12010d001601
0d05090c-040f-191d-0a13-12010d061601
0d05090c-040f-191d-0a13-12010d091601
0d05090c-040f-191d-0a13-12010d0b1601
0d05090c-040f-191d-0a13-12010d0c0001
0d05090c-040f-191d-0a13-12010d0c0b01
0d05090c-040f-191d-0a13-12010d0c1101
0d05090c-040f-191d-0a13-12010d0c1401
0d05090c-040f-191d-0a13-12010d0c1501
0d05090c-040f-191d-0a13-12010d0c1600
.
=== Outcome ===
071f0c18-1a01-1610-0519-021b0a081615
=== Shrinks ===
071f0c00-1a01-1610-0519-021b0a081615
071f0c0c-1a01-1610-0519-021b0a081615
071f0c12-1a01-1610-0519-021b0a081615
071f0c15-1a01-1610-0519-021b0a081615
071f0c17-1a01-1610-0519-021b0a081615
071f0018-1a01-1610-0519-021b0a081615
071f0618-1a01-1610-0519-021b0a081615
071f0918-1a01-1610-0519-021b0a081615
071f0b18-1a01-1610-0519-021b0a081615
07000c18-1a01-1610-0519-021b0a081615
07100c18-1a01-1610-0519-021b0a081615
07180c18-1a01-1610-0519-021b0a081615
071c0c18-1a01-1610-0519-021b0a081615
071e0c18-1a01-1610-0519-021b0a081615
001f0c18-1a01-1610-0519-021b0a081615
041f0c18-1a01-1610-0519-021b0a081615
061f0c18-1a01-1610-0519-021b0a081615
071f0c18-1a00-1610-0519-021b0a081615
071f0c18-0001-1610-0519-021b0a081615
071f0c18-0d01-1610-0519-021b0a081615
071f0c18-1401-1610-0519-021b0a081615
071f0c18-1701-1610-0519-021b0a081615
071f0c18-1901-1610-0519-021b0a081615
071f0c18-1a01-1600-0519-021b0a081615
071f0c18-1a01-1608-0519-021b0a081615
071f0c18-1a01-160c-0519-021b0a081615
071f0c18-1a01-160e-0519-021b0a081615
071f0c18-1a01-160f-0519-021b0a081615
071f0c18-1a01-0010-0519-021b0a081615
071f0c18-1a01-0b10-0519-021b0a081615
071f0c18-1a01-1110-0519-021b0a081615
071f0c18-1a01-1410-0519-021b0a081615
071f0c18-1a01-1510-0519-021b0a081615
071f0c18-1a01-1610-0019-021b0a081615
071f0c18-1a01-1610-0319-021b0a081615
071f0c18-1a01-1610-0419-021b0a081615
071f0c18-1a01-1610-0500-021b0a081615
071f0c18-1a01-1610-050d-021b0a081615
071f0c18-1a01-1610-0513-021b0a081615
071f0c18-1a01-1610-0516-021b0a081615
071f0c18-1a01-1610-0518-021b0a081615
071f0c18-1a01-1610-0519-001b0a081615
071f0c18-1a01-1610-0519-011b0a081615
071f0c18-1a01-1610-0519-02000a081615
071f0c18-1a01-1610-0519-020e0a081615
071f0c18-1a01-1610-0519-02150a081615
071f0c18-1a01-1610-0519-02180a081615
071f0c18-1a01-1610-0519-021a0a081615
071f0c18-1a01-1610-0519-021b00081615
071f0c18-1a01-1610-0519-021b05081615
071f0c18-1a01-1610-0519-021b08081615
071f0c18-1a01-1610-0519-021b09081615
071f0c18-1a01-1610-0519-021b0a001615
071f0c18-1a01-1610-0519-021b0a041615
071f0c18-1a01-1610-0519-021b0a061615
071f0c18-1a01-1610-0519-021b0a071615
071f0c18-1a01-1610-0519-021b0a080015
071f0c18-1a01-1610-0519-021b0a080b15
071f0c18-1a01-1610-0519-021b0a081115
071f0c18-1a01-1610-0519-021b0a081415
071f0c18-1a01-1610-0519-021b0a081515
071f0c18-1a01-1610-0519-021b0a081600
071f0c18-1a01-1610-0519-021b0a08160b
071f0c18-1a01-1610-0519-021b0a081610
071f0c18-1a01-1610-0519-021b0a081613
071f0c18-1a01-1610-0519-021b0a081614
.
=== Outcome ===
020b080a-1616-0d11-041d-051705191c17
=== Shrinks ===
020b0800-1616-0d11-041d-051705191c17
020b0805-1616-0d11-041d-051705191c17
020b0808-1616-0d11-041d-051705191c17
020b0809-1616-0d11-041d-051705191c17
020b000a-1616-0d11-041d-051705191c17
020b040a-1616-0d11-041d-051705191c17
020b060a-1616-0d11-041d-051705191c17
020b070a-1616-0d11-041d-051705191c17
0200080a-1616-0d11-041d-051705191c17
0206080a-1616-0d11-041d-051705191c17
0209080a-1616-0d11-041d-051705191c17
020a080a-1616-0d11-041d-051705191c17
000b080a-1616-0d11-041d-051705191c17
010b080a-1616-0d11-041d-051705191c17
020b080a-1600-0d11-041d-051705191c17
020b080a-160b-0d11-041d-051705191c17
020b080a-1611-0d11-041d-051705191c17
020b080a-1614-0d11-041d-051705191c17
020b080a-1615-0d11-041d-051705191c17
020b080a-0016-0d11-041d-051705191c17
020b080a-0b16-0d11-041d-051705191c17
020b080a-1116-0d11-041d-051705191c17
020b080a-1416-0d11-041d-051705191c17
020b080a-1516-0d11-041d-051705191c17
020b080a-1616-0d00-041d-051705191c17
020b080a-1616-0d09-041d-051705191c17
020b080a-1616-0d0d-041d-051705191c17
020b080a-1616-0d0f-041d-051705191c17
020b080a-1616-0d10-041d-051705191c17
020b080a-1616-0011-041d-051705191c17
020b080a-1616-0711-041d-051705191c17
020b080a-1616-0a11-041d-051705191c17
020b080a-1616-0c11-041d-051705191c17
020b080a-1616-0d11-001d-051705191c17
020b080a-1616-0d11-021d-051705191c17
020b080a-1616-0d11-031d-051705191c17
020b080a-1616-0d11-0400-051705191c17
020b080a-1616-0d11-040f-051705191c17
020b080a-1616-0d11-0416-051705191c17
020b080a-1616-0d11-041a-051705191c17
020b080a-1616-0d11-041c-051705191c17
020b080a-1616-0d11-041d-001705191c17
020b080a-1616-0d11-041d-031705191c17
020b080a-1616-0d11-041d-041705191c17
020b080a-1616-0d11-041d-050005191c17
020b080a-1616-0d11-041d-050c05191c17
020b080a-1616-0d11-041d-051205191c17
020b080a-1616-0d11-041d-051505191c17
020b080a-1616-0d11-041d-051605191c17
020b080a-1616-0d11-041d-051700191c17
020b080a-1616-0d11-041d-051703191c17
020b080a-1616-0d11-041d-051704191c17
020b080a-1616-0d11-041d-051705001c17
020b080a-1616-0d11-041d-0517050d1c17
020b080a-1616-0d11-041d-051705131c17
020b080a-1616-0d11-041d-051705161c17
020b080a-1616-0d11-041d-051705181c17
020b080a-1616-0d11-041d-051705190017
020b080a-1616-0d11-041d-051705190e17
020b080a-1616-0d11-041d-051705191517
020b080a-1616-0d11-041d-051705191917
020b080a-1616-0d11-041d-051705191b17
020b080a-1616-0d11-041d-051705191c00
020b080a-1616-0d11-041d-051705191c0c
020b080a-1616-0d11-041d-051705191c12
020b080a-1616-0d11-041d-051705191c15
020b080a-1616-0d11-041d-051705191c16
.
=== Outcome ===
1b191619-0c1d-0f19-1d20-060803021912
=== Shrinks ===
1b191600-0c1d-0f19-1d20-060803021912
1b19160d-0c1d-0f19-1d20-060803021912
1b191613-0c1d-0f19-1d20-060803021912
1b191616-0c1d-0f19-1d20-060803021912
1b191618-0c1d-0f19-1d20-060803021912
1b190019-0c1d-0f19-1d20-060803021912
1b190b19-0c1d-0f19-1d20-060803021912
1b191119-0c1d-0f19-1d20-060803021912
1b191419-0c1d-0f19-1d20-060803021912
1b191519-0c1d-0f19-1d20-060803021912
1b001619-0c1d-0f19-1d20-060803021912
1b0d1619-0c1d-0f19-1d20-060803021912
1b131619-0c1d-0f19-1d20-060803021912
1b161619-0c1d-0f19-1d20-060803021912
1b181619-0c1d-0f19-1d20-060803021912
00191619-0c1d-0f19-1d20-060803021912
0e191619-0c1d-0f19-1d20-060803021912
15191619-0c1d-0f19-1d20-060803021912
18191619-0c1d-0f19-1d20-060803021912
1a191619-0c1d-0f19-1d20-060803021912
1b191619-0c00-0f19-1d20-060803021912
1b191619-0c0f-0f19-1d20-060803021912
1b191619-0c16-0f19-1d20-060803021912
1b191619-0c1a-0f19-1d20-060803021912
1b191619-0c1c-0f19-1d20-060803021912
1b191619-001d-0f19-1d20-060803021912
1b191619-061d-0f19-1d20-060803021912
1b191619-091d-0f19-1d20-060803021912
1b191619-0b1d-0f19-1d20-060803021912
1b191619-0c1d-0f00-1d20-060803021912
1b191619-0c1d-0f0d-1d20-060803021912
1b191619-0c1d-0f13-1d20-060803021912
1b191619-0c1d-0f16-1d20-060803021912
1b191619-0c1d-0f18-1d20-060803021912
1b191619-0c1d-0019-1d20-060803021912
1b191619-0c1d-0819-1d20-060803021912
1b191619-0c1d-0c19-1d20-060803021912
1b191619-0c1d-0e19-1d20-060803021912
1b191619-0c1d-0f19-0020-060803021912
1b191619-0c1d-0f19-0f20-060803021912
1b191619-0c1d-0f19-1620-060803021912
1b191619-0c1d-0f19-1a20-060803021912
1b191619-0c1d-0f19-1c20-060803021912
1b191619-0c1d-0f19-1d00-060803021912
1b191619-0c1d-0f19-1d10-060803021912
1b191619-0c1d-0f19-1d18-060803021912
1b191619-0c1d-0f19-1d1c-060803021912
1b191619-0c1d-0f19-1d1e-060803021912
1b191619-0c1d-0f19-1d1f-060803021912
1b191619-0c1d-0f19-1d20-000803021912
1b191619-0c1d-0f19-1d20-030803021912
1b191619-0c1d-0f19-1d20-050803021912
1b191619-0c1d-0f19-1d20-060003021912
1b191619-0c1d-0f19-1d20-060403021912
1b191619-0c1d-0f19-1d20-060603021912
1b191619-0c1d-0f19-1d20-060703021912
1b191619-0c1d-0f19-1d20-060800021912
1b191619-0c1d-0f19-1d20-060802021912
1b191619-0c1d-0f19-1d20-060803001912
1b191619-0c1d-0f19-1d20-060803011912
1b191619-0c1d-0f19-1d20-060803020012
1b191619-0c1d-0f19-1d20-060803020d12
1b191619-0c1d-0f19-1d20-060803021312
1b191619-0c1d-0f19-1d20-060803021612
1b191619-0c1d-0f19-1d20-060803021812
1b191619-0c1d-0f19-1d20-060803021900
1b191619-0c1d-0f19-1d20-060803021909
1b191619-0c1d-0f19-1d20-06080302190e
1b191619-0c1d-0f19-1d20-060803021910
1b191619-0c1d-0f19-1d20-060803021911
.
=== Outcome ===
0a1c000b-0b1b-0020-110b-0a0a15010f18
=== Shrinks ===
0a1c0000-0b1b-0020-110b-0a0a15010f18
0a1c0006-0b1b-0020-110b-0a0a15010f18
0a1c0009-0b1b-0020-110b-0a0a15010f18
0a1c000a-0b1b-0020-110b-0a0a15010f18
0a00000b-0b1b-0020-110b-0a0a15010f18
0a0e000b-0b1b-0020-110b-0a0a15010f18
0a15000b-0b1b-0020-110b-0a0a15010f18
0a19000b-0b1b-0020-110b-0a0a15010f18
0a1b000b-0b1b-0020-110b-0a0a15010f18
001c000b-0b1b-0020-110b-0a0a15010f18
051c000b-0b1b-0020-110b-0a0a15010f18
081c000b-0b1b-0020-110b-0a0a15010f18
091c000b-0b1b-0020-110b-0a0a15010f18
0a1c000b-0b00-0020-110b-0a0a15010f18
0a1c000b-0b0e-0020-110b-0a0a15010f18
0a1c000b-0b15-0020-110b-0a0a15010f18
0a1c000b-0b18-0020-110b-0a0a15010f18
0a1c000b-0b1a-0020-110b-0a0a15010f18
0a1c000b-001b-0020-110b-0a0a15010f18
0a1c000b-061b-0020-110b-0a0a15010f18
0a1c000b-091b-0020-110b-0a0a15010f18
0a1c000b-0a1b-0020-110b-0a0a15010f18
0a1c000b-0b1b-0000-110b-0a0a15010f18
0a1c000b-0b1b-0010-110b-0a0a15010f18
0a1c000b-0b1b-0018-110b-0a0a15010f18
0a1c000b-0b1b-001c-110b-0a0a15010f18
0a1c000b-0b1b-001e-110b-0a0a15010f18
0a1c000b-0b1b-001f-110b-0a0a15010f18
0a1c000b-0b1b-0020-000b-0a0a15010f18
0a1c000b-0b1b-0020-090b-0a0a15010f18
0a1c000b-0b1b-0020-0d0b-0a0a15010f18
0a1c000b-0b1b-0020-0f0b-0a0a15010f18
0a1c000b-0b1b-0020-100b-0a0a15010f18
0a1c000b-0b1b-0020-1100-0a0a15010f18
0a1c000b-0b1b-0020-1106-0a0a15010f18
0a1c000b-0b1b-0020-1109-0a0a15010f18
0a1c000b-0b1b-0020-110a-0a0a15010f18
0a1c000b-0b1b-0020-110b-000a15010f18
0a1c000b-0b1b-0020-110b-050a15010f18
0a1c000b-0b1b-0020-110b-080a15010f18
0a1c000b-0b1b-0020-110b-090a15010f18
0a1c000b-0b1b-0020-110b-0a0015010f18
0a1c000b-0b1b-0020-110b-0a0515010f18
0a1c000b-0b1b-0020-110b-0a0815010f18
0a1c000b-0b1b-0020-110b-0a0915010f18
0a1c000b-0b1b-0020-110b-0a0a00010f18
0a1c000b-0b1b-0020-110b-0a0a0b010f18
0a1c000b-0b1b-0020-110b-0a0a10010f18
0a1c000b-0b1b-0020-110b-0a0a13010f18
0a1c000b-0b1b-0020-110b-0a0a14010f18
0a1c000b-0b1b-0020-110b-0a0a15000f18
0a1c000b-0b1b-0020-110b-0a0a15010018
0a1c000b-0b1b-0020-110b-0a0a15010818
0a1c000b-0b1b-0020-110b-0a0a15010c18
0a1c000b-0b1b-0020-110b-0a0a15010e18
0a1c000b-0b1b-0020-110b-0a0a15010f00
0a1c000b-0b1b-0020-110b-0a0a15010f0c
0a1c000b-0b1b-0020-110b-0a0a15010f12
0a1c000b-0b1b-0020-110b-0a0a15010f15
0a1c000b-0b1b-0020-110b-0a0a15010f17
.
val it : unit = ()
> 
```